pynamodb/expressions/util.py

```diff
@@ -0,0 +1,29 @@
+import re
+
+PATH_SEGMENT_REGEX = re.compile(r'([^\[\]]+)((?:\[\d+\])*)$')
+
+
+def substitute_names(expression, placeholders):
+    """
+    Replaces names in the given expression with placeholders.
+    Stores the placeholders in the given dictionary.
+    """
+    path_segments = expression.split('.')
```

Comment:
Currently it wouldn't do the right thing -- I hope to address this in my next changeset.